Source/GSCSStrength.m

```diff
@@ -0,0 +1,112 @@
+/* Copyright (C) 2023 Free Software Foundation, Inc.
+
+   By: Benjamin Johnson
+   Date: 19-3-2023
+   This file is part of the GNUstep Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "GSCSStrength.h"
+#import "GSCSFloatComparator.h"
+
+@implementation GSCSStrength
+
+- (instancetype) initWithName: (NSString *)name strength: (double)strength;
+{
+  if (self = [super init])
+    {
+      ASSIGN(_name, name);
+      _strength = strength;
+    }
+  return self;
+}
+
++ (instancetype) strengthRequired
+{
+  return AUTORELEASE([[GSCSStrength alloc] initWithName: @"<Required>" strength: 1000]);
```

Comment:
Yes, technically you could share strength objects but it would add a decent amount of complexity, every constraint owning its own strength keeps it simple

Reply:
It depends on how many of these objects will be used in a typical application. And I don't see the added complexity. Just add a static variable for each method, initialise it with nil and if it is set return that value. Otherwise create the object like the current code does and set the static variable to it. About five lines per method.